### DISCUSSION TITLE
Bootstrap 4

### DISCUSSION THREAD

**Discussion by matheusggds** [frontendbr/forum#541] [Perguntas e Respostas]
Olá Frontenderos,

Então gostaria de saber o que vocês acham sobre o Bootstrap 4, por mais que esteja na versão Alpha ( Alpha 6 ), já é possível utiliza-lo e muito bem (na minha opnião).

Estou achando essa integração com o SASS incrível, um arquivo só de variáveis onde você pode customizar praticamente tudo que já vem como padrão, ativar radius, ativar shadow, ativar o flexbox e muitos mais... Existem classes para tudo, praticamente tudo que você desejar fazer em relação a ajustes do GRID e Layout para mobile, é possível através de classes já prontas.

Você só precisar realmente escrever código CSS em elementos totalmente novos, ou para mudar o estilo de algum já existente. Fora a tipografia responsiva ( não que isso seja uma super novidade, mas eu particularmente só comecei a usa-lo agora com o Bootstrap 4 pois o formato de tamanho das fonts utilizando [REM](http://desenvolvimentoparaweb.com/css/unidades-css-rem-vh-vw-vmin-vmax-ex-ch/) facilita muito ).

Eu utilizei o bootstrap 3 por uns 3 anos, e mudei para o 4 a mais ou menos 5 meses, e estou achando muito bom e atendendo a tudo que necessito. Você consegue customizar tudo no layout de forma prática e rápida.

Alguém aqui já utiliza? Está esperando a versão oficial lançar, algo a acrescentar? 

Abraços.

**Comment by RodolfoSilva** [comment]
Eu já utilizo com Angular 2/4, cara eu estou achando muito bal.... O mais legal de tudo é que acho que ficou mais organizado e mais facil de customizar.

**Comment by eduardojmatos** [comment]
Ainda não utilizei, mas o esquema de grids com flexbox parece legal. Estamos usando grids com Flexbox no [gaiden](https://github.com/getninjas/gaiden/) - já que pra IE9 aplicamos um CSS específico com correções.

**Comment by torto** [comment]
Eu gosto muito do Bootstrap 4, flexbos é muito amor e até hoje nunca tive problemas com ele nas aplicações que uso!

**Comment by felipefialho** [comment]
Não sou hater do Bootstrap, pelo contrário, amo esse framework, tive um pull-request aprovado em 2013 e acho que ele foi essencial para consolidação de algumas boas práticas atuais.

Mas tirando caso especificos, como demonstrações, CMS ou necessidade pontual de desenvolvedores back-ends com poucas skills em front-end, não vejo necessidade nenhuma em usar frameworks (isso vale para Foundation e todo o resto também).

Vejo muitas pessoas ainda usando por causa dos grids mas existem soluções baseadas em PostCSS (funciona com qualquer coisa) não intrusivas, que usam flexbox faz anos como o [Lost](https://github.com/peterramsing/lost).

É mais dificil adaptar (e principalmente dar manutenção futura) os componentes que vem nesses frameworks enormes, do que criar aplicações pensadas do zero.

**Comment by thulioph** [comment]
Não utilizei essa nova versão do Bootstrap, parei na 3.7 e sinceramente não sinto nenhuma falta. 

Antes era bem comum utiliza-lo por conta do fator desempenho, onde eu conseguia entregar coisas 
bem rápidas de uma maneira simples e que várias pessoas conseguiam entender, seja com freelas, demos, workshops ou qualquer coisa que precisava ser rápido.

Desde que comecei a estudar Vue, vi muito material diferente utilizando um carinha chamado [Bulma](http://bulma.io/) e até então só utilizo ele nos meus projetos. O que mais me chamou atenção é que ele não utiliza absolutamente nada de JS, deixando tudo na responsabilidade do CSS.

Quer alterar o estado de algum elemento? Classes como: `is-active` ou `is-open`, podem ajudar você a fazer o toggle via Javascript do jeito que você quiser, sem te prender a marcação do Framework.. Precisa trabalhar com grids? Algumas alternativas foram ditas anteriormente, mas o bulma também possui uma. Além de uma série de componentes já prontos..

Claro, perto do Bootstrap o Bulma é apenas um menino, mas esse menino tem me ajudado e muito, espero que seja útil para alguém 😄 

> Acredito que no final tudo vai depender do que você precisa e de qual o seu objetivo.

**Comment by kvnol** [comment]
Compartilho da mesma opinião do @LFeh, como disse no Twitter:

> O Bootstrap está para facilitar o trampo dos back-enders, assim como o WordPress está para facilitar o trampo dos front-enders.

**Comment by ghost** [comment]
Venho usado somente a grid e gostado muito! Recomendo!

**Comment by bernardocs** [comment]
Compartilho totalmente minha opinião com @LFeh ...

Grid Layout tá aí pra acabar com esses frameworks. Digo, não acabar né... Estarão todos aí... Mas sem necessidade nenhuma, já que o Grid Layout supre, tecnicamente, a parte de layout quase toda.

Aliás... muito embora flexbox esteja aí a **muitos** muitos anos e a galera mal usa ainda. Espero muito que o Grid seja bem disseminado. 

Vamos fazer o possível! :smile:

**Comment by ghost** [comment]
Eu sou um adepto de grid pronta, isso me ajuda e muito na hora de estruturar o site e na hora de fazer uma futura manutenção.

Gosto de separar a estrutura dos componentes, colocar as colunas em seus lugares como o site foi diagramado e depois venho criando os componentes. Como se fosse uma construção de um imóvel.

Utilizei por um tempo o _Jeet_ e depois o _Lost_ e desde o começo eu sentia que me dava mto trabalho pra fazer as duas coisas ao mesmo tempo. Sem dizer de manutenção para aumentar ou diminuir uma área, que foi facilmente solucionado apenas mudando uma classe sem precisar compilar nada.

Atualmente uso a grid do bootstrap 4 em flexblox, compilei apenas o arquivo específico para isso.

**Comment by didiraja** [comment]
MUITISSIMO OBRIGADO a todos desse topico. Graças a ele conheci o Bulma e parei pra olhar o Bootstrap v4 q parece sensacional. Evitava por estar no alpha ainda mas está com recursos maravilhosos, essa integração com flexbox é tudo que precisava haha

**Comment by kvnol** [comment]
@didiraja flexbox é amor. <3

**Comment by yurimatheus** [comment]
Há um 5 anos uso o Bootstrap nos meus projetos e ele tem atendido todas as
necessidades de cada projeto.

O flexbox ficou sensacional realmente, acredito que meu maior problema em
usar outro framework é o uso das grids que ao meu ver no Bootstrap é bem
fácil até pelo uso e constância.

Quanto a outros frameworks front o Bulma realmente é muito bom, tem o Ui
Kit e o Semantic Ui que são grandes projetos também e podem bater de frente
com o B4.

Em 24 de mai de 2017 12:39 AM, "Kevin Oliveira" <notifications@github.com>
escreveu:

**Comment by expalmer** [comment]
Bootstrap foi e ainda é muito importante para comunidade, o problema é que essa geração de devs de agora, não estão se dando o trabalho de fazer as coisas por si mesmo. E além disso, usam bootstrap para projetos pessoais tanto para projetos em produção...acabando não entendo como funcionam as coisas.
Vida longa aos css frameworks, mas não se esqueça de fazer você mesmo.

**Comment by joseliacosta** [comment]
Hey @all resolvi dar um *revival* nessa issue porque estou muito intrigada com a maneira que o bootstrap 4 está tratando algumas coisas, então não sei se é uma dúvida que compensa abrir outra issue ou se é má interpretação minha da documentação.

Vocês que estão utilizando o bootstrap 4, tem conseguido usar o component `card` e tratar manualmente a largura do `card` com as colunas que o framework disponibiliza?

**Comment by felipefialho** [comment]
@joselialcosta Se quiser abrir uma issue não tem problema não viu.

Essa issue é bem ampla.
